pallets/remote-asset-manager/src/mock.rs

```diff
@@ -220,15 +352,73 @@ pub mod para {
 
     parameter_types! {
         pub LockupPeriod: <Runtime as system::Config>::BlockNumber = 10;
-        pub MinimumRedemption: u32 = 2;
+        pub MinimumRedemption: u32 = 0;
         pub WithdrawalPeriod: <Runtime as system::Config>::BlockNumber = 10;
         pub DOTContributionLimit: Balance = 999;
         pub TreasuryPalletId: PalletId = PalletId(*b"12345678");
-        pub StringLimit: u32 = 4;
+        pub PalletStringLimit: u32 = 4;
+
+        pub const RelayChainAssetId: AssetId = RELAY_CHAIN_ASSET;
+        pub const PINTAssetId: AssetId = PARA_ASSET;
+        pub SelfLocation: MultiLocation = MultiLocation::X2(Junction::Parent, Junction::Parachain(ParachainInfo::parachain_id().into()));
+    }
+
+    ord_parameter_types! {
+        pub const AdminAccountId: AccountId = ADMIN_ACCOUNT;
+    }
+
+    impl pallet_asset_index::Config for Runtime {
+        // Using signed as the admin origin for testing now
+        type AdminOrigin = frame_system::EnsureSigned<AccountId>;
+        type Event = Event;
+        type AssetId = AssetId;
+        type IndexToken = Balances;
+        type Balance = Balance;
+        type LockupPeriod = LockupPeriod;
+        type MinimumRedemption = MinimumRedemption;
+        type WithdrawalPeriod = WithdrawalPeriod;
+        type DOTContributionLimit = DOTContributionLimit;
+        type RemoteAssetManager = MockRemote;
+        type Currency = Currency;
+        type PriceFeed = MockPriceFeed;
+        type TreasuryPalletId = TreasuryPalletId;
+        type WithdrawalFee = ();
+        type StringLimit = PalletStringLimit;
+        type WeightInfo = ();
+    }
+
+    pub struct MockRemote;
+    use sp_runtime::DispatchResult;
+    impl pallet_remote_asset_manager::traits::RemoteAssetManager<AccountId, u32, u128> for MockRemote {
```

Comment:
```suggestion
    use pallet_remote_asset_manager::traits::RemoteAssetManager as RemoteAssetManagerTrait;
    impl pallet_remote_asset_manager::traits::RemoteAssetManager<AccountId, u32, u128> for MockRemote {
```

Reply:
still does not compile -.-
makes no sense whatsoever